text/0079-exception-groups.md

```diff
@@ -0,0 +1,558 @@
+  - Its value is always the same as `InnerExceptions[0]`.
+
+## Python
+
+The exception group type in .Python is [`ExceptionGroup`](https://docs.python.org/3/library/exceptions.html#exception-groups).
+- The exceptions of the group are stored in the `exceptions` attribute.
+- Like other exceptions, it can have a `__cause__` and/or a `__context__` attribute.
+  - `__context__` is an indirect cause, assigned if another exception occurs while handling the exception.
+  - `__cause__` is a direct cause, assigned if raised with the exception (using the `from` keyword).
+    - Setting it suppresses any `__context__` value, when displayed in the stack trace.
+  - There is no requirement that `exceptions[0]` be either of these.
+
+## JavaScript
+
+The exception group type in JavaScript is [`AggregateError`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/AggregateError).
+- The errors of the group are stored in the `errors` property.
+- Like other errors, it has a `cause` (singular) property.
+  - There is no requirement that `errors[0]` be the same as `cause`.
+
+# Proposed Solution
+
+We will break down the problem into two phases, as described in the following sections.
+
+## Phase 1
+
+In this phase, we will focus on having the "primary issue" of the exception group surfaced in Sentry in a meaningful way.
+Instead of representing every possible exception in the group, we will build an issue by walking only the first path through
+the exception tree.  In other words, SDKs will follow the chain from the captured exception to its _first_ inner exception,
+repeatedly until there are no more inner exceptions.  As with the current Sentry design, these exceptions will be sent in the
+[`exception`](https://develop.sentry.dev/sdk/event-payloads/exception/) interface of the event message - ordered from oldest to newest.
+
+As an example, if the exception tree is:
+
+- A
+  - B
+    - C
+      - D
+  - E
+    - F
+      - G
+
+The corresponding issues sent in the `exceptions` interface would be `[D, C, B, A]` in that order.
+The issue title would be based on exception `A`, and issue grouping would consider exceptions `A`, `B`, `C`, and `D` only.
+
+Exceptions `E`, `F`, and `G` _should not_ be sent in the `exceptions` interface.  The reasons for this are:
+- They may be representing completely different issues that shouldn't be grouped together in Sentry.
+- They may be representing different quantities of the _same_ issue, and the quantity should not affect issue grouping in Sentry either.
+- Sending them all in one list would imply a causal relationship between `D` and `E` (in that example), which does not exist.
+
+However, we don't want to completely lose track of these other exceptions either.  Thus they should still be sent
+with the event, but in another section instead.  We will add it to the `extras` property of the event.
+
+The format should be as follows:
+
+```json
+"extras": {
```

Comment:
Yes, it should be `extra`, not `extras`.  I will fix that - thanks.  As far as I can tell, it is not deprecated.  Is there somewhere that says it has been?

`debug_meta` would not be visible in the UI presently.  We should pick some place that the data is visible even for older versions of self-hosted Sentry.  I also think it could affect processing in unexpected ways.

`contexts` would be ideal, but it has half of the [size limit](https://develop.sentry.dev/sdk/data-handling/#variable-size) of `extra`, which I can see being an issue for larger exception groups - especially if they contain stack traces.

Reply:
Update: I just spent some time discussing this with @mitsuhiko, and he agrees that we should _not_ use `extras`.

I'm going to explore yet another alternative, which is to use `exceptions` with new fields on `mechanism`.  I'll update the RFC with the results, if that works out.

Thanks.

Reply:
Major rewrite complete.  Have another read.  Thanks. 😉